src/events/balancer.py

```diff
@@ -34,7 +36,7 @@ def __init__(self, charm) -> None:
         self.workload = BalancerWorkload()
 
         # Fast exit after workload instantiation, but before any event observer
-        if BALANCER.value not in self.charm.config.roles:
+        if BALANCER.value not in self.charm.config.roles or not self.charm.unit.is_leader():
```

Comment:
This change prevents non-leader units from writing CC config files. Not that big of a deal, but it avoids cluttering the logs